crates/oci-tar-builder/src/lib.rs

```diff
@@ -10,14 +10,76 @@ use oci_spec::image::{
     DescriptorBuilder, ImageConfiguration, ImageIndexBuilder, ImageManifestBuilder, MediaType,
     PlatformBuilder, SCHEMA_VERSION,
 };
+use oci_wasm::{WasmConfig, WASM_ARCHITECTURE};
 use serde::Serialize;
 use sha256::{digest, try_digest};
-#[derive(Debug, Default)]
-pub struct Builder {
-    configs: Vec<(ImageConfiguration, String)>,
+#[derive(Debug)]
+pub struct Builder<C: OciConfig> {
+    configs: Vec<(C, String, MediaType)>,
     layers: Vec<(PathBuf, String)>,
 }
 
+pub trait OciConfig {
+    fn os(&self) -> String;
+    fn architecture(&self) -> String;
+    fn layers(&self) -> Vec<String>;
+    fn to_string(&self) -> String;
+}
```

Comment:
suggestion:

```suggestion
pub trait OciConfig {
    fn os(&self) -> &str;
    fn architecture(&self) -> &str;
    fn layers(&self) -> &[String];
}

impl<T: OciConfig> ToString for T {
    fn to_string(&self) -> String {
        serde_json::to_string_pretty(self).unwrap()
    }
}
```

Reply:
I tried this out but I get various errors like:

```
define and implement a trait or new type insteadrustc[Click for full compiler diagnostic](rust-analyzer-diagnostics-view:/diagnostic%20message%20%5B0%5D?0#file%3A%2F%2F%2Fhome%2Fjstur%2Fprojects%2Frunwasi%2Fcrates%2Foci-tar-builder%2Fsrc%2Flib.rs)
lib.rs(43, 19): `WasmConfig` is not defined in the current crate
```

```
| impl<T: OciConfig> ToString for T {
   |      ^ type parameter `T` must be used as the type parameter for some local type
```

I think this is again an issue with the fact that I am extending types outside of the current crate